src/utils/deploy/deploy-site.js

```diff
@@ -66,6 +67,18 @@ const deploySite = async (
     throw new Error('No files or functions to deploy')
   }
 
+  if (functionsWithNativeModules.length !== 0) {
+    warn(`Modules with native dependencies\n
+${functionsWithNativeModules.map(({ name }) => `- ${name}`).join('\n')}    
+
+These functions use Node.js modules with native dependencies, which must be installed
+on a system with the same architecture as the runtime. Failing to do so might lead to
+errors when invoking your functions.
+To ensure your functions work as expected, we recommend using continuous deployment.
+
+For more information, visit https://ntl.fyi/cli-native-modules.`)
```

Comment:
I haven't hooked up this link yet, but I was thinking of pointing it to https://docs.netlify.com/site-deploys/create-deploys/#deploy-with-git.